compliance/features/esql.feature

```diff
@@ -0,0 +1,8 @@
+Feature: ES|QL
+  Features related to ES|QL
+
+  @3.4.2
```

Comment:
Could you please replace 3.4.1 with 3.4.2 in these lines? This way this scenario will be tested in CI. https://github.com/elastic/package-spec/blob/69f3a8bd718ba3060b5eb8e6ad7e5f2701db56d3/.buildkite/pipeline.trigger.compliance.tests.sh#L31-L32